### PR TITLE
Enable properties which supported in sdk

### PR DESCRIPTION
**Pull request by zhichengliu12581**
enable properties which supported in sdk
related [pr](https://github.com/Azure/azure-sdk-for-java/pull/22604)

### CODE REVIEW

File: sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml
Comment (stliu) [inline sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml:22]: is this added new?
very wired

Comment (zhichengliu12581) [reply sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml:22]: Yes, in order to add an integration test for the configuration of the added `serviceBusReceiveMode` parameter, I can delete it if it is not necessary😅

**Comment by stliu** [comment]
docs?

**Comment by zhichengliu12581** [comment]
@stliu I'm waiting for the review for this [pr](https://github.com/Azure/azure-sdk-for-java/pull/22778), which have conflict with this pr. I plan to modify the document and changelog after that pr merge. Or I can add it temporarily, and then modify it again after the pr merge?

File: sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml
Comment (stliu) [reply sdk/spring/azure-spring-cloud-test-servicebus-binder/src/test/resources/application-properties.yml:22]: never mind, I was wrong, just asked Yi Liu about this

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusProperties.java
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusProperties.java:21]: Is this default value for servicebus sdk client? If so, seems we don't set it here

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusTopicAutoConfiguration.java
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusTopicAutoConfiguration.java:75]: how about using setRetryOptions to keep all function names consistent?

File: sdk/spring/azure-spring-cloud-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusAutoConfigurationTest.java
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusAutoConfigurationTest.java:101]: Can we add UTs in the queue&topic autoconfigurtion to see if the client factory beans are created as the way the properties are configured?

File: sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md:52]: Seems we have a default value?

File: sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md:119]: Should we add introdution for maxConcurrentCalls as well?

Comment (zhichengliu12581) [reply sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md:119]: It already has an introduction to maxConcurrentCalls [here](https://github.com/zhichengliu12581/azure-sdk-for-java/blob/dc8756197ea3c09fec4caa9215933105bc5eebab/sdk/spring/azure-spring-cloud-stream-binder-servicebus-queue/README.md#L94)

File: sdk/spring/azure-spring-cloud-autoconfigure/CHANGELOG.md
Comment (backwind1233) [inline sdk/spring/azure-spring-cloud-autoconfigure/CHANGELOG.md:14]: Deprecated --> Deprecate

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusQueueAutoConfiguration.java
Comment (backwind1233) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusQueueAutoConfiguration.java:67]: can we delete the empty line here?

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusTopicAutoConfiguration.java
Comment (backwind1233) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/servicebus/AzureServiceBusTopicAutoConfiguration.java:76]: can we delete the empty line here?

**Review by yiliuTo — APPROVED** [review]
lgtm

**Pull request merged**
